corehq/apps/api/odata/serializers.py

```diff
@@ -71,3 +73,73 @@ def _clean_property_name(name):
                     properties.pop(property_name)
 
         return json.dumps(data, cls=DjangoJSONEncoder, sort_keys=True)
+
+
+class ODataXFormInstanceSerializer(Serializer):
+    """
+    A custom serializer that converts form data into an odata-compliant format.
+    Must be paired with ODataXFormInstanceResource
+    """
+    def to_json(self, data, options=None):
+        options = options or {}
+
+        domain = data.pop('domain', None)
+        if not domain:
+            raise Exception('API requires domain to be set! Did you add it in a custom create_response function?')
```

Comment:
Should these be some subclass of `Exception` so they can be handled separately from totally unexpected errors?

Reply:
replaced with assertions